configure.ac

```diff
@@ -476,15 +476,9 @@ esac
 ## Data types and their sizes.
 ##
 AC_TYPE_OFF_T
-AC_CHECK_TYPE([size_t], [],
-        [AC_DEFINE_UNQUOTED([size_t], [unsigned long],
-                [Define to `unsigned long' if <sys/types.h> does not define.])])
 AC_CHECK_TYPE([ssize_t], [],
         [AC_DEFINE_UNQUOTED([ssize_t], [long],
                 [Define to `long' if <sys/types.h> does not define.])])
-AC_CHECK_TYPE([ptrdiff_t], [],
-        [AC_DEFINE_UNQUOTED([ptrdiff_t], [long],
-                [Define to `long' if <sys/types.h> does not define.])])
```

Comment:
These checks are useless because the types are decorated with the H5_ prefix and we don't use H5_size_t, etc. in the library. size_t and ptrdiff_t are in the C standard anyway, so they are doubly useless. I'm leaving ssize_t and off_t alone for now since those are POSIX and we may want to strip the H5_ and use this definition instead of H5public.h.